File: OpenCV/15_orb/cpp/main.cpp
#include <opencv2/opencv.hpp>
#include <iostream>
int main(){ auto g=cv::imread("input.jpg",cv::IMREAD_GRAYSCALE); if(g.empty()) return 1; auto o=cv::ORB::create(1000,1.2f,8); std::vector<cv::KeyPoint> k; cv::Mat d,out; o->detectAndCompute(g,cv::noArray(),k,d); cv::drawKeypoints(g,k,out); std::cout<<k.size()<<" keypoints\n"; cv::imwrite("output_orb.png",out); }

File: OpenCV/15_orb/python/main.py
import cv2
g=cv2.imread("input.jpg",cv2.IMREAD_GRAYSCALE); assert g is not None
orb=cv2.ORB_create(nfeatures=1000,scaleFactor=1.2,nlevels=8); kp,des=orb.detectAndCompute(g,None); out=cv2.drawKeypoints(g,kp,None); print(len(kp),des.shape if des is not None else None); cv2.imwrite("output_orb.png",out)
